lib/cell.rb

```diff
@@ -25,18 +25,22 @@ def fired_upon?
   def fire_upon
     @fired_upon = true
     if empty? == false
-      ship.hit
+      @ship.hit
     else
       "M"
     end
   end
 
   def render(show_ship = false)
-    return "S" if show_ship && @ship
+    return "S" if show_ship && @ship unless @fired_upon == true
     return "." unless @fired_upon
     return "M" if @fired_upon == true && empty?
-    return "X" if @ship.sunk?
-    "H"
+    return "X" if sunk?
+    return "H"
+  end
+
+  def sunk?
+    @ship.sunk? if @ship!=nil
```

Comment:
i wonder what is more time efficient regarding computation and memory if the if staement came prior to @ship.sunk? Please give me your thoughts...